l10n_th_tin_service/models/res_partner.py

```diff
@@ -0,0 +1,196 @@
+        sess = Session()
+        mod_dir = path.dirname(path.realpath(__file__))
+        cert_path = str(Path(mod_dir).parents[0]) + "/static/cert/adhq1_ADHQ5.cer"
+        sess.verify = cert_path
+        transp = Transport(session=sess)
+        try:
+            cl = Client(ResPartner.vat_web_service_url, transport=transp)
+        except requests.exceptions.SSLError:
+            _logger.log(INFO, "Fall back to unverifed HTTPS request.")
+            sess.verify = False
+            transp = Transport(session=sess)
+            cl = Client(ResPartner.vat_web_service_url, transport=transp)
+        result = cl.service.Service(
+            "anonymous",
+            "anonymous",
+            TIN=tin,
+            ProvinceCode=0,
+            BranchNumber=branch,
+            AmphurCode=0,
+        )
+        odata = helpers.serialize_object(result)
+        _logger.log(DEBUG, pprint.pformat(odata))
+        data = OrderedDict()
+        if odata["vmsgerr"] is None:
+            for key, value in odata.items():
+                if (
+                    value is None
+                    or value["anyType"][0] == "-"
+                    or key in {"vNID", "vBusinessFirstDate"}
+                ):
+                    continue
+                data[key] = value["anyType"][0]
+            return data
+        return False
+
+    @api.onchange("vat", "branch")
+    def _onchange_vat_branch(self):
+        word_map = {
+            "vBuildingName": "อาคาร ",
+            "vFloorNumber": "ชั้นที่ ",
+            "vVillageName": "หมู่บ้าน ",
+            "vRoomNumber": "ห้องเลขที่ ",
+            "vHouseNumber": "เลขที่ ",
+            "vMooNumber": "หมู่ที่ ",
+            "vSoiName": "ซอย ",
+            "vStreetName": "ถนน ",
+            "vThambol": "ต.",
+            "vAmphur": "อ.",
+            "vProvince": "จ.",
+            "vPostCode": "",
+        }
+        street_map = [
+            [
+                "vBuildingName",
+                "vRoomNumber",
+                "vFloorNumber",
+                "vHouseNumber",
+                "vStreetName",
+                "vSoiName",
+            ],
+            ["vVillageName", "vMooNumber", "vThambol"],
+        ]
+        check_branch = re.compile(r"^\d{5}$")
+        if self.branch is False:
+            self.branch = "00000"
+
+        if self.vat:
+            if ResPartner.check_rd_tin_service(self.vat):
+                match = check_branch.match(self.branch)
+                if match is None:
+                    return {
+                        "warning": {
+                            "title": _("Branch validation failed"),
+                            "message": _(
+                                "Branch number %s must be 5 digits." % self.branch
+                            ),
+                        }
+                    }
+                data = ResPartner.get_info_rd_vat_service(self.vat, self.branch)
+                _logger.log(DEBUG, pprint.pformat(data))
```

Comment:
In case of an error from calling the API, the `data` is `False`.  `_logger.log(DEBUG, "False")` does not make any sense for debugging. Please put `data = _get_info_rd_vat_service(self.vat, self.branch)` in a `try...except...` block.
```suggestion
                try:
                    data = ResPartner.get_info_rd_vat_service(self.vat, self.branch)
                    _logger.log(DEBUG, pprint.pformat(data))
                except:
                    _logger.log(DEBUG, _("There is an error calling RD's API"))  # or other suitable error message
```